### PR TITLE
feat(web): Album preview overview in menu

### PR DESCRIPTION
**Pull request by TimVanOnckelen**
This PR introduces a visual overview of the three latest albums in the main menu, similar to the App-Which-Must-Not-Be-Named. The addition provides quick album previews for users, enhancing navigation and user experience.

### Changes Overview:

Album Menu Extension: Added a visual preview of three albums directly within the Album menu for quick access.
New Optional Parameter in API: Introduced top as an optional parameter in the /albums (getAllAlbums) route, allowing flexible limits on album retrieval.
Technical Details:

The top parameter supports optimized retrieval of album data by limiting results, enhancing efficiency for both UI rendering and API load.
The visual display integrates seamlessly with the existing menu styling, ensuring consistent user experience.

image

### CODE REVIEW

**Comment by bo0tzz** [comment]
Implements part of #8743

**Comment by alextran1502** [comment]
Hello, thanks for the PR. I am working on migrate the whole web codebase to the new rune syntax so new features on the web will have to wait 😅

**Comment by TimVanOnckelen** [comment]
> Hello, thanks for the PR. I am working on migrate the whole web codebase to the new rune syntax so new features on the web will have to wait 😅

Np, ofcourse! Where possible, i used your changes on the rune syntax from your pr #13883

**Comment by TimVanOnckelen** [comment]
I will temporary set this PR back to draft until the svelte 5 migration is done.

File: web/package-lock.json
Comment (danieldietzler) [inline web/package-lock.json:1]: What happened to this file?

**Comment by alextran1502** [comment]
We don't need to modify the server query. We can use the existing getAllAlbums, in the future we can optimize this behavior by having a dedicated endpoint to search for albums

**Comment by TimVanOnckelen** [comment]
> We don't need to modify the server query. We can use the existing getAllAlbums, in the future we can optimize this behavior by having a dedicated endpoint to search for albums

fair point, i'll remove the changes and just add a filter on the frontend.

**Comment by TimVanOnckelen** [comment]
I removed all server changes, the PR is now web only.

File: web/src/lib/components/shared-components/side-bar/side-bar-link.svelte [hunk 1]
```diff
@@ -51,6 +55,24 @@
...
   </div>
 
+  {#if hasDropdown}
+    <span class="hidden md:block">
+      <button
+        type="button"
+        class="relative flex cursor-default select-none justify-center hover:cursor-pointer hover:bg-immich-gray hover:text-immich-primary dark:text-immich-dark-fg dark:hover:bg-immich-dark-gray dark:hover:text-immich-dark-primary rounded"
+        onclick={() => (dropdownOpen = !dropdownOpen)}
+      >
+        <Icon
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/side-bar-link.svelte:65]: What do you think about putting this icon on the left side? It feels strange to use by having it on the right side and the tooltip blocks it when you hover over the `i` icon.

File: web/src/lib/components/shared-components/side-bar/side-bar.svelte [hunk 2]
```diff
@@ -102,10 +103,15 @@
       {/snippet}
     </SideBarLink>
 
-    <SideBarLink title={$t('albums')} routeId="/(user)/albums" icon={mdiImageAlbum} flippedLogo>
+    <SideBarLink title={$t('albums')} routeId="/(user)/albums" icon={mdiImageAlbum} flippedLogo dropdownOpen>
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/side-bar.svelte:106]: Maybe persist the `dropdownOpen` state to localstorage

File: web/src/lib/components/shared-components/side-bar/recent-albums.svelte [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  {/each}
+  <a
+    href="/albums"
+    aria-label="albums"
+    class="flex w-full place-items-center justify-between gap-4 rounded-r-full py-3 transition-[padding] delay-100 duration-100 hover:cursor-pointer hover:bg-immich-gray hover:text-immich-primary dark:text-immich-dark-fg dark:hover:bg-immich-dark-gray dark:hover:text-immich-dark-primary pl-10 group-hover:sm:px-10 md:px-10"
+  >
+    <div>
+      <div class="h-6 w-6 bg-cover rounded"><Icon path={mdiArrowRight}></Icon></div>
+    </div>
+    <div class="grow text-sm font-medium">{$t('all_albums')}</div>
```
[the remark this answers is not in the document]
Comment (TimVanOnckelen) [reply web/src/lib/components/shared-components/side-bar/recent-albums.svelte:50]: good point. It's seems better to remove it

File: web/src/lib/components/shared-components/side-bar/side-bar-link.svelte (hunk 1 above)
Comment (TimVanOnckelen) [reply web/src/lib/components/shared-components/side-bar/side-bar-link.svelte:95]: What about this?
image

File: web/src/lib/components/shared-components/side-bar/recent-albums.svelte [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    try {
+      const allAlbums = await getAllAlbums({});
+      albums = allAlbums.slice(0, 3);
+    } catch (error) {
+      handleError(error, $t('failed_to_load_assets'));
+    }
+  });
+</script>
+
+{#if albums.length > 0}
```
Comment (danieldietzler) [inline web/src/lib/components/shared-components/side-bar/recent-albums.svelte:20]: You don't need this `if`. If albums is empty the for each loop will simply have no iteration, resulting in the same

File: web/src/lib/components/shared-components/side-bar/recent-albums.svelte [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  import { getAllAlbums, type AlbumResponseDto } from '@immich/sdk';
+  import { handleError } from '$lib/utils/handle-error';
+  import { t } from 'svelte-i18n';
+
+  let albums: AlbumResponseDto[] = $state([]);
+
+  onMount(async () => {
+    try {
+      const allAlbums = await getAllAlbums({});
+      albums = allAlbums.slice(0, 3);
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/recent-albums.svelte:13]: The albums are currently sorted by ascending creation date. How about sorting them by most recently updated instead?

File: web/src/lib/components/shared-components/side-bar/recent-albums.svelte [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+{#each albums as album}
+  <a
+    href={'/albums/' + album.id}
+    title={album.albumName}
+    class="flex w-full place-items-center justify-between gap-4 rounded-r-full py-3 transition-[padding] delay-100 duration-100 hover:cursor-pointer hover:bg-immich-gray hover:text-immich-primary dark:text-immich-dark-fg dark:hover:bg-immich-dark-gray dark:hover:text-immich-dark-primary pl-10 group-hover:sm:px-10 md:px-10"
+  >
+    <div>
+      <div
+        class="h-6 w-6 bg-cover rounded"
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/recent-albums.svelte:28]: A gray background for missing thumbnails looks more consistent

```suggestion
        class="h-6 w-6 bg-cover rounded bg-gray-200 dark:bg-gray-600"
```

File: web/src/lib/components/shared-components/side-bar/side-bar-link.svelte [hunk 7]
```diff
@@ -34,49 +38,71 @@
...
+              </div>
+            </div>
+          {/if}
+        </div>
+      {/if}
+    </div>
+  </a>
+  {#if hasDropdown}
+    <span class="hidden md:block absolute top-4 left-1">
+      <button
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/side-bar-link.svelte:90]: - This button doesn't display correctly in firefox
- It should have an accessible name
- Maybe increase the clickable area, at least vertically

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**